### PR TITLE
[fa] day 1

### PR DESCRIPTION
**Pull request by alix1383**
- **sync #2038 pr**
- **add #2127 changes**

### CODE REVIEW

**Review by DannyRavi — APPROVED** [review]
there is no Error.

File: po/fa.po [hunk 1]
```diff
@@ -1809,44 +1750,42 @@ msgstr ""
...
 
 #: src/running-the-course/course-structure.md:21
 msgid "Day 1 Afternoon (2 hours and 35 minutes, including breaks)"
-msgstr ""
+msgstr "روز اول بعدازظهر (2 ساعت و 35 دقیقه با احتساب استراحت)"
 
 #: src/running-the-course/course-structure.md:25 src/welcome-day-1-afternoon.md
 msgid "35 minutes"
-msgstr ""
+msgstr "35 دقیقه"
```
Comment (DannyRavi) [inline po/fa.po:1768]: بهتره از کیبورد استاندارد فارسی استفاده بشه و اعادا به فارسی باشه
مثلا ۳۵ و نه 35

File: po/fa.po [hunk 2]
```diff
@@ -3767,22 +3902,22 @@ msgstr ""
 
 #: src/welcome-day-1-afternoon.md src/welcome-day-2-afternoon.md
 #: src/welcome-day-3-afternoon.md src/welcome-day-4-afternoon.md
-#, fuzzy
 msgid "Welcome Back"
-msgstr "خوش آمدید"
+msgstr "خوش آمد"
```
Comment (DannyRavi) [inline po/fa.po:3906]: خوش آمدید درست تره
بهتره ترجمه کلمه به کلیمه نباشه و گر چیز به فارسی متداول هست اون معنی متداول رو بیاری

File: po/fa.po [hunk 3]
```diff
@@ -3767,22 +3902,22 @@ msgstr ""
...
 "Including 10 minute breaks, this session should take about 2 hours and 35 "
 "minutes. It contains:"
 msgstr ""
+"با احتساب 10 دقیقه استراحت، این جلسه باید حدود 2 ساعت و 35 دقیقه طول بکشد. "
+"آن شامل:"
 
 #: src/tuples-and-arrays.md
 msgid "This segment should take about 35 minutes. It contains:"
-msgstr ""
+msgstr "این بخش باید حدود 35 دقیقه طول بکشد. آن شامل:"
```
Comment (DannyRavi) [inline po/fa.po:3918]: این شاملِ

**Review by DannyRavi — CHANGES_REQUESTED** [review]
Thank you for the translations, please, the average translations should be about 100 lines or translations so that the translated text can be reviewed better.

File: po/fa.po [hunk 4]
```diff
@@ -169,83 +163,77 @@ msgstr "توابع"
...
 #: src/SUMMARY.md
 msgid "Day 1: Afternoon"
-msgstr "روز ۱: بعد از ظهر"
+msgstr "روز اول: بعد از ظهر"
 
 #: src/SUMMARY.md src/running-the-course/course-structure.md:25
 #: src/welcome-day-1-afternoon.md src/tuples-and-arrays.md
 msgid "Tuples and Arrays"
-msgstr ""
+msgstr "چندتایی‌ها(Tuples) و آرایه‌ها"
```
[the remark this answers is not in the document]
Comment (alix1383) [reply po/fa.po:180]: یکی از دوستان گفتن چنتایی بهتره و چون قراره به فارسی باشه همون چنتایی بنظرم اوکی باشه

https://github.com/google/comprehensive-rust/pull/2038#discussion_r1631251197

**Comment by alix1383** [comment]
از اینکه وقت گذاشتین ممنونم @DannyRavi 
مورد اول دلیل حذف شدن بعضی از عبارات کن کنم msgmerge بوده باشه برام خودمم عجیب بود 
مورد بعدی اینکه با توجه به توصیه های شما بهتره که 2 بخش اول رو دوباره چک کنم و ترجمه مناسب تری رو اراعه بدم البته بعد از merge  شدن این pr

**Comment by DannyRavi** [comment]
@alix1383 
املای نانوشته بدون خطا و ایراد هست.
لطفاً موارد خطا را برطرف کنید تا این pull request رو merge کنم.
همینطور بهتر هست مبحث مورد ترجمه در آینده رو انتخاب و reserve کنید، تا با افراد دیگه که روی این پروژه کار می‌کنند، تداخل نداشته باشید.
Good luck.

**Review by DannyRavi — APPROVED** [review]
now it's ok.

**Comment by DannyRavi** [comment]
@alix1383 please resolve 
Error formatting /home/runner/work/comprehensive-rust/comprehensive-rust/po/fa.po. Message: Child process exited with code 1: <stdin>:20355: duplicate message definition...

File: package.json
Comment (DannyRavi) [inline package.json:1]: just use `git add ` on fa.po file. never  use `git add` and pushed any other files.

**Review by DannyRavi — CHANGES_REQUESTED** [review]
just push fa.po

**Review by DannyRavi — APPROVED** [review]

File: po/fa.po [hunk 5]
```diff
@@ -83,28 +82,27 @@ msgstr ""
...
+msgstr "تایپ و مقدارها"
 
 #: src/SUMMARY.md src/types-and-values.md src/types-and-values/variables.md:1
 msgid "Variables"
 msgstr "متغیرها"
 
 #: src/SUMMARY.md src/types-and-values.md src/types-and-values/values.md:1
 msgid "Values"
-msgstr ""
+msgstr "مقدارها"
```
Comment (moaminsharifi) [inline po/fa.po:93]: I think It's better to use "مقادیر" for it.  in computer science normally we use "مقادیر".

File: po/fa.po [hunk 6]
```diff
@@ -169,83 +163,77 @@ msgstr "توابع"
...
+msgstr "چندتایی‌ها(Tuples) و آرایه‌ها"
 
 #: src/SUMMARY.md src/tuples-and-arrays.md src/tuples-and-arrays/arrays.md:1
 msgid "Arrays"
 msgstr "آرایه‌ها"
 
 #: src/SUMMARY.md src/tuples-and-arrays.md src/tuples-and-arrays/tuples.md:1
 msgid "Tuples"
-msgstr "تاپل‌ها"
+msgstr "چندتایی‌ها"
```
[the remark this answers is not in the document]
Comment (moaminsharifi) [reply po/fa.po:188]: I agree, normally we use "تاپل" for "tuple"

File: po/fa.po [hunk 7]
```diff
@@ -191,61 +185,55 @@ msgstr "آرایه‌ها"
...
 #: src/SUMMARY.md src/running-the-course/course-structure.md:26
 #: src/welcome-day-1-afternoon.md src/references.md
 msgid "References"
 msgstr "مراجع"
 
 #: src/SUMMARY.md src/references.md src/references/shared.md:1
-#, fuzzy
 msgid "Shared References"
-msgstr "مراجع"
+msgstr "مراجع مشترک"
```
Comment (moaminsharifi) [inline po/fa.po:210]: It's a Reference which shared between two things. so "اشتراکی" is better for it.

File: po/fa.po [hunk 8]
```diff
@@ -191,61 +185,55 @@ msgstr "آرایه‌ها"
...
-#, fuzzy
 msgid "Exclusive References"
-msgstr "ارجاعات تعلیق شده"
+msgstr "مراجع انحصاری  "
 
 #: src/SUMMARY.md
-#, fuzzy
 msgid "Slices: `&[T]`"
-msgstr "برش‌ها"
+msgstr "برش ها: `&[T]`"
```
Comment (moaminsharifi) [inline po/fa.po:218]: I think I will render badly with considering #1413 
Also Please use half-space between "برش" and "ها" like "برش‌ها"

File: po/fa.po [hunk 9]
```diff
@@ -282,11 +269,11 @@ msgstr "روز دوم: صبح"
 #: src/SUMMARY.md src/running-the-course/course-structure.md:35
 #: src/welcome-day-2.md src/pattern-matching.md
 msgid "Pattern Matching"
-msgstr "تطبیق الگو"
```
Comment (moaminsharifi) [inline po/fa.po:285]: I think "تطبیق" is better for it, dude to other translation in the computer science.

File: po/fa.po [hunk 10]
```diff
@@ -282,11 +269,11 @@ msgstr "روز دوم: صبح"
...
+msgstr "تطابق الگو"
 
 #: src/SUMMARY.md src/pattern-matching.md src/pattern-matching/match.md:1
 msgid "Matching Values"
-msgstr ""
+msgstr "تطابق مقدارها"
```
Comment (moaminsharifi) [inline po/fa.po:276]: I think " تطبیق مقادیر" is better for it, dude to other translation in the computer science.

File: po/fa.po [hunk 11]
```diff
@@ -255,7 +243,7 @@ msgstr "ساختار‌های چند‌گانه (Tuple Structs)"
 #: src/SUMMARY.md src/user-defined-types.md src/user-defined-types/enums.md:1
 #: src/pattern-matching/destructuring-enums.md:1
 msgid "Enums"
-msgstr "Enums"
+msgstr "مقادیر شمارشی"
```
Comment (moaminsharifi) [inline po/fa.po:246]: For consistency, I recommend not translating 'Enum' throughout the book.

File: po/fa.po [hunk 12]
```diff
@@ -295,23 +282,21 @@ msgstr "تخریب ساختارها"
 #: src/SUMMARY.md src/pattern-matching.md
 #, fuzzy
 msgid "Destructuring Enums"
-msgstr "تخریب Enumها"
```
Comment (moaminsharifi) [inline po/fa.po:298]: due to line 254 comment I think "Enum‌های تفکیک‌پذیر" is better for it.
It's not like old-school `destructure C` function, I more like "[Destructuring assignment in JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment)"

File: po/fa.po [hunk 13]
```diff
@@ -295,23 +282,21 @@ msgstr "تخریب ساختارها"
...
+msgstr "تخریب ساختار Enums"
 
 #: src/SUMMARY.md src/pattern-matching.md
 #: src/pattern-matching/let-control-flow.md:1
-#, fuzzy
 msgid "Let Control Flow"
-msgstr "کنترل جریان"
+msgstr "کنترل جریان (Let)"
```
Comment (moaminsharifi) [inline po/fa.po:290]: Please don't use `(` and `)` just `Let`

File: po/fa.po [hunk 14]
```diff
@@ -322,28 +307,25 @@ msgid "Traits"
...
 #: src/methods-and-traits/deriving.md:1
-#, fuzzy
 msgid "Deriving"
-msgstr "صفات استنتاجی (Deriving Traits)"
+msgstr ""
 
 #: src/SUMMARY.md src/methods-and-traits.md
 msgid "Exercise: Generic Logger"
-msgstr ""
+msgstr "توابع جنریک"
```
Comment (moaminsharifi) [inline po/fa.po:328]: Add "تمرین"  to start of string

File: po/fa.po [hunk 15]
```diff
@@ -387,32 +368,29 @@ msgstr ""
 
 #: src/SUMMARY.md src/running-the-course/course-structure.md:44
 #: src/welcome-day-2-afternoon.md src/std-types.md
-#, fuzzy
 msgid "Standard Library Types"
-msgstr "کتابخانه استاندارد انواع داده"
+msgstr "کتابخانه‌ استاندارد تایپ ها"
```
Comment (moaminsharifi) [inline po/fa.po:372]: Please don't change "انواع داده" to "تایپ ها"

File: po/fa.po [hunk 16]
```diff
@@ -540,66 +510,57 @@ msgstr ""
...
-#, fuzzy
 msgid "Exercise: Health Statistics"
-msgstr "آمارهای سلامتی"
+msgstr "تمرین: آمار سلامتی"
 
 #: src/SUMMARY.md src/running-the-course/course-structure.md:62
 #: src/welcome-day-3-afternoon.md src/lifetimes.md
-#, fuzzy
 msgid "Lifetimes"
-msgstr "چرخه حیات"
```
Comment (moaminsharifi) [inline po/fa.po:578]: I think "چرخه حیاط" make more sense for it, because it has  `start .... > end`

File: po/fa.po [hunk 17]
```diff
@@ -2081,17 +2022,17 @@ msgstr "چندین میانبر صفحه کلید مفید در mdBook وجود
...
 #: src/running-the-course/keyboard-shortcuts.md:6
 #, fuzzy
 msgid "<kbd>Arrow-Right</kbd>: Navigate to the next page."
-msgstr "به صفحه بعدی هدایت می کند"
+msgstr "به صفحه بعدی هدایت می کند."
 
 #: src/running-the-course/keyboard-shortcuts.md:7
 #, fuzzy
 msgid "<kbd>Ctrl + Enter</kbd>: Execute the code sample that has focus."
-msgstr "اجرای نمونه کدی که بر روی آن قرار گرفته است."
```
Comment (moaminsharifi) [inline po/fa.po:2094]: where is `<kbd>Ctrl + Enter</kbd>`?

File: po/fa.po [hunk 18]
```diff
@@ -2617,13 +2654,15 @@ msgstr "تطابق الگو: تجزیه و تحلیل enums, structs و آرای
 #: src/welcome-day-4.md src/concurrency/welcome.md
 #: src/concurrency/welcome-async.md
 msgid "Schedule"
-msgstr ""
+msgstr "برنامه"
```
Comment (moaminsharifi) [inline po/fa.po:2657]: I am not sure "برنامه" which use for "application" is a good choice for it.

**Review by moaminsharifi — CHANGES_REQUESTED** [review]
After nearly 10 months away, I'm excited to be returning to active contributions. I aim to review up to 3000 pull requests.

Please avoid deleting translations and consider leaving them empty as suggested by @DannyRavi.

Thank you for your contributions.

**Comment by DannyRavi** [comment]
@alix1383 I think you `commit` a part of content with different git username or email in your local machine or use another machine to commit and pull request. so cla/google has problem with your pull request and show `Missing CLA from one or more contributors`.

**Comment by DannyRavi** [comment]
me and @javad-jafari started this project when there was no Persian language development for this project.
Fortunately, after some time, new people volunteered for the Persian part of this project, and Javad and I thought that it would be better for the new volunteers to continue and finish the translation.

But now in my opinion, without coordination and cooperation with each other, the Persian translation of this book may not be successful.

I suggest that we have a channel on Telegram and a [project](https://github.com/orgs/farsi-tools/projects/2/views/1) on GitHub to manage and clarify the tasks and do them to finish this book.
We do not get results without organization!

@javad-jafari  @alix1383 @moaminsharifi @hamidrezakp  @mehrad77
what is your opinion?

File: po/fa.po (hunk 16 above)
Comment (javad-jafari) [reply po/fa.po:578]: but "حیاط" means Yard

File: po/fa.po (hunk 18 above)
Comment (javad-jafari) [reply po/fa.po]: the "زمان بندی" is better

**Comment by moaminsharifi** [comment]
> me and @javad-jafari started this project when there was no Persian language development for this project. Fortunately, after some time, new people volunteered for the Persian part of this project, and Javad and I thought that it would be better for the new volunteers to continue and finish the translation.
> 
> But now in my opinion, without coordination and cooperation with each other, the Persian translation of this book may not be successful.
> 
> I suggest that we have a group on Telegram and a [project](https://github.com/orgs/farsi-tools/projects/2/views/1) on GitHub to manage and clarify the tasks and do them to finish this book. We do not get results without organization!
> 
> @javad-jafari @alix1383 @moaminsharifi @hamidrezakp @mehrad77 what is your opinion?


I agree that translation progress has been slow. I started contributing to the translation about a year ago, but the lack of reviewers has significantly impacted the process.

I'm eager to help accelerate the translation process. Let's create a plan to streamline the translation workflow.

**Comment by alix1383** [comment]
image
این صفحه به درستی رندر نمیشه به (فارسی ترجمه نمیشه)

**Comment by DannyRavi** [comment]
> > me and @javad-jafari started this project when there was no Persian language development for this project. Fortunately, after some time, new people volunteered for the Persian part of this project, and Javad and I thought that it would be better for the new volunteers to continue and finish the translation.
> > But now in my opinion, without coordination and cooperation with each other, the Persian translation of this book may not be successful.
> > I suggest that we have a group on Telegram and a [project](https://github.com/orgs/farsi-tools/projects/2/views/1) on GitHub to manage and clarify the tasks and do them to finish this book. We do not get results without organization!
> > @javad-jafari @alix1383 @moaminsharifi @hamidrezakp @mehrad77 what is your opinion?
> 
> I agree that translation progress has been slow. I started contributing to the translation about a year ago, but the lack of reviewers has significantly impacted the process.
> 
> I'm eager to help accelerate the translation process. Let's create a plan to streamline the translation workflow.

I don't think the problem is with the reviewers.
In my opinion, for cooperation and successful completion of this project, this group for this project on [Telegram](https://t.me/+HAKbn2dKIqQ3NTQ0) and this [board](https://github.com/orgs/farsi-tools/projects/2/views/1) on GitHub can be effective.
Please [join this group](https://t.me/+HAKbn2dKIqQ3NTQ0).
Thanks.

@javad-jafari @alix1383 @moaminsharifi @hamidrezakp @mehrad77

**Pull request closed without merging**